src/main/scala/ru/otus/sc/game/dao/impl/GameProcessDaoImpl.scala

```diff
@@ -0,0 +1,347 @@
+    val mapWithEnemies = GameMap(
+      List
+        .range(0, 5)
+        .foldLeft(localMap.entities)((acc, _) => {
+          acc + (freeCell(acc) -> Player.enemy())
+        })
+    )
+    val pos = freeCell(mapWithEnemies.entities)
+    this.gameState.get(player.nick) match {
+      case Some(gameState) =>
+        this.gameState(player.nick) = gameState.copy(
+          player = player.copy(health = PlayerStats.maxHealth),
+          map = mapWithEnemies.entities,
+          state = GameProcessState.MAIN_MENU,
+          pos = pos
+        )
+      case None =>
+        this.gameState.put(
+          player.nick,
+          PlayerCurrentGameState(player, GameProcessState.MAIN_MENU, mapWithEnemies.entities, pos)
+        )
+    }
+
+    val state = this.gameState(player.nick)
+    this.gsd.save(state.player, state.map, state.pos)
+
+    true
+  }
+
+  /**
+    * Загрузка игры
+    *
+    * @param player игрок
+    * @return смог или не смог
+    */
+  override def loadGame(player: Player): Boolean = {
+    this.gameState.get(player.nick) match {
+      case Some(gameState) =>
+        val savedMap    = this.gsd.loadMap(player)
+        val savedPlayer = this.gsd.loadPlayer(player)
+
+        if (savedMap.isDefined && savedPlayer.isDefined) {
+          this.gameState(player.nick) = gameState.copy(
+            player = savedPlayer.get._1,
+            pos = savedPlayer.get._2,
+            map = savedMap.get
+          )
+          true
+        } else {
+          false
+        }
+      case None =>
+        false
+    }
+  }
+
+  /**
+    * Сохранение игры
+    *
+    * @param player игрок
+    * @return смог или не смог
+    */
+  override def saveGame(player: Player): Boolean = {
+    this.gameState.get(player.nick) match {
+      case Some(gameState) =>
+        this.gsd.savePlayer(gameState.player, gameState.pos) && this.gsd
+          .saveMap(gameState.player, gameState.map)
+      case None =>
+        false
+    }
+  }
+
+  /**
+    * Игрок заходит в игру.
+    * Надо сделать поиск сохранения. Если сохранения нет - новая игра
+    *
+    * @param player игрок
+    * @return
+    */
+  override def signIn(player: Player): Boolean = {
+    this.gameState.get(player.nick) match {
+      case Some(gameState) =>
+        this.gameState(player.nick) = gameState.copy(state = GameProcessState.MAIN_MENU)
+      case None =>
+        this.gameState.put(
+          player.nick,
+          PlayerCurrentGameState(player, GameProcessState.MAIN_MENU, Map.empty, Position(0, 0))
+        )
+    }
+    true
+  }
+
+  /**
+    * Игрок заходит на карту.
+    * После входа на карту - меняется меню на игровое.
+    *
+    * @param player игрок
+    * @return смог или не смог зайти на карту
+    */
+  override def enterMap(player: Player): Boolean = {
+    this.gameState.get(player.nick) match {
+      case Some(gameState) =>
+        this.gameState(gameState.player.nick) = gameState.copy(state = GameProcessState.GAME)
+      case None =>
```

Comment:
```suggestion
      case None => ()
```

Лучше всегда явно указывать что нам не важно значение.